commit e8bb480ea9b9cac675c5ab510f4eabcb8ecce375
Author: Alexis Ballier <aballier@gentoo.org>
Date:   Thu May 22 17:49:01 2008 +0200

    Revert "Remove the forced ld -z,defs flags."
    
    This reverts commit 074ed6708644323bd441332fb7d34b99bd954350.
    
    Useful to avoid runtime errors, it's better to spot them at build time.

Index: vlc-0.9.0-git/configure.ac
===================================================================
--- vlc-0.9.0-git.orig/configure.ac
+++ vlc-0.9.0-git/configure.ac
@@ -984,6 +984,15 @@ CFLAGS="${CFLAGS_save}"
 RDC_PROG_CC_WFLAGS([all extra sign-compare undef pointer-arith bad-function-cast cast-align write-strings missing-prototypes volatile-register-var])
 RDC_PROG_CC_FLAGS_IFELSE([-pipe])
 
+dnl This helps detect broken plugins with undefined symbols
+dnl Better fail loudly at build-time than silenty at run-time.
+dnl Unfortunately, this sucks on FreeBSD (won't even link with -lc)
+if test `uname -s` != FreeBSD; then
+  RDC_PROG_LINK_FLAGS_IFELSE([-Wl,-z,defs], [
+    VLC_ADD_LDFLAGS([plugin],[-Wl,-z,defs])
+  ])
+fi
+
 dnl Check for various optimization flags
 AC_CACHE_CHECK([if \$CC accepts -Os],
     [ac_cv_c_os],
